### PR TITLE
fix `align_bottom` widgets erasing

### LINKED ISSUE #1616: View border and status bar background overlap when vertical widget layout is enabled
Woof version: 14.3.0
Vertical Layout: ON
vertical_layout_on

Vertical Layout: OFF
vertical_layout_off

### PR DESCRIPTION
**Pull request by rfomin**
Fix #1616

### CODE REVIEW

File: src/hu_lib.c [hunk 1]
```diff
@@ -519,7 +519,10 @@ void HUlib_erase_widget (const hu_widget_t *const w)
   const hu_font_t *const f = *m->font;
 
   const int height = m->numlines * f->line_height;
-  const int y = vert_align_widget(w, m, f, w->h_align, w->v_align);
+  int y = vert_align_widget(w, m, f, w->h_align, w->v_align);
```
Comment (fabiangreffrath) [inline src/hu_lib.c:522]: Hm, I think you need to call `vert_align_widget()` `m->numlines` times. Else the next widget after the multiline widget will not get properly cleaned.

Comment (rfomin) [reply src/hu_lib.c]: Done in https://github.com/fabiangreffrath/woof/pull/1628/commits/5f7e4683dbbabecc099b3f6ed8476d8e65ca2538

**Comment by fabiangreffrath** [comment]
I think it needs this:

```patch
--- a/src/hu_lib.c
+++ b/src/hu_lib.c
@@ -526,6 +526,9 @@ void HUlib_erase_widget (const hu_widget_t *const w)
     y = vert_align_widget(w, m, f, w->h_align, w->v_align);
   }
 
+  if (w->v_align == align_top)
+    y -= height;
+
   if (y > scaledviewy && y < scaledviewy + scaledviewheight - height)
   {
     R_VideoErase(0, y, scaledviewx, height);
```

**Comment by rfomin** [comment]
> ```diff
> +  if (w->v_align == align_top)
> +    y -= height;
> ```

I have tried this, but the bug returns. The current version seems to work correctly.

**Comment by fabiangreffrath** [comment]
In the current version the bug has moved to top aligned widgets.

**Comment by rfomin** [comment]
> In the current version the bug has moved to top aligned widgets.

Right, sorry, I misread the patch. Fixed in: https://github.com/fabiangreffrath/woof/pull/1628/commits/52bb338c4db4d02115dc979d546f766e4d58e76d

**Review by fabiangreffrath — APPROVED** [review]
I think that's it now. Thank you!

**Pull request merged**